### DISCUSSION TITLE
How to give better performance, hybrid OpenMP/MPI or pure MPI?

### DISCUSSION THREAD

**Discussion by lo-y-wni** [ufs-community/ufs-srweather-app#733] [Q&A] [answered]
Hello everyone:

I'm curious that how to set the numbers of MPI/OpenMP can give best performance. I'm currently using AWS EC2 and totally have 4 nodes, 144 processors for my experiments. To best utilize the power of EC2, I wish to find the best combination of MPI/OpenMP to give the most efficient setting. Can someone share his/her experience of configuring the numbers of MPI and OpenMP?

Yi-Chuan

**Comment by EdwardSnyder-NOAA** [comment, marked as answer]
Good afternoon Yi-Chuan,

Given that you are using 144 total processors, we expect little performance variations using different MPI/OpenMP configurations. We tend to see more performance variations or improvements between MPI/OpenMP configurations when using 1,000 processors or more. Our test experiments tend to have a consistent configuration based on the experiments grid, which can be found here: `ufs-srweather-app/ush/predef_grid_params.yaml`. For your situation, any valid configuration would be sufficient. But if you are interested in finding the most efficient settings, that would require running a series of experiments with different configurations and comparing the results. Also, keep in mind your EC2 costs as well when running your experiments. That could help dictate your experiments configuration too.

Please let us know if you have any additional questions.

Thanks!

Eddie Snyder | Advanced User Support (AUS) Team
